### PR TITLE
Add meta tags and social share fields

### PR DESCRIPTION
**Pull request by arh30**
I've added a meta include to use the variables defined in settings or on the page itself - @richpeters got this working for CinW and I think it's a good pattern to go with! 

Question for BE: should I be reusing the `share_image` and `share_description` fields on the pages, or do they have to be added for each page type?

### CODE REVIEW

File: {{cookiecutter.project_slug}}/apps/pages/models.py [hunk 1]
```diff
@@ -10,8 +13,61 @@ class StandardPage(Page):
...
+        related_name="+",
+        help_text="""
+        Image shown when this page is shared on social media.
+        """,
+    )
+
+    # Promote panels
+    promote_panels = Page.promote_panels + [
+        MultiFieldPanel(
+            [FieldPanel("share_description"), ImageChooserPanel("share_image")],
```
Comment (richpeters) [inline {{cookiecutter.project_slug}}/apps/pages/models.py]: Can we add a trailing comma to the list, might be easier to read quickly

File: {{cookiecutter.project_slug}}/apps/settings/models.py [hunk 2]
```diff
@@ -39,3 +40,47 @@ class FooterHTML(BaseSetting):
...
     class Meta:
         verbose_name = "HTML - Footer"
+
+
+@register_setting(icon="link")
+class SocialAccounts(BaseSetting):
+    facebook_url = models.URLField(
+        blank=True,
+        help_text="""
+        The URL of your organisation\'s Facebook page. E.g.
```
Comment (richpeters) [inline {{cookiecutter.project_slug}}/apps/settings/models.py]: This shouldn't need to be escaped in multi-line string

File: {{cookiecutter.project_slug}}/apps/settings/models.py [hunk 3]
```diff
@@ -39,3 +40,47 @@ class FooterHTML(BaseSetting):
...
+        "https://www.facebook.com/organisation_name_or_id"'
+        """,
+    )
+    twitter_handle = models.URLField(
```
Comment (richpeters) [inline {{cookiecutter.project_slug}}/apps/settings/models.py]: I think it's better to use twitter url as the URL may change over time

Comment (arh30) [reply {{cookiecutter.project_slug}}/apps/settings/models.py]: I think then we can change that in the code, and have control over the url used. The twitter handle is useful in the twitter meta tags to link the shared posts to the orgs twitter. Do you think we need an extra field for url 🤔

File: {{cookiecutter.project_slug}}/apps/settings/models.py (hunk 2 above)
Comment (arh30) [reply {{cookiecutter.project_slug}}/apps/settings/models.py]: do you mean it should be one line, or just using apostrophes and not speech marks?

File: {{cookiecutter.project_slug}}/apps/pages/models.py (hunk 1 above)
Comment (arh30) [reply {{cookiecutter.project_slug}}/apps/pages/models.py]: would that be `FieldPanel("share_description"), ImageChooserPanel("share_image"),],`?

**Comment by alistairclark** [comment]
RE the question about `share_image` and `share_description`: you could create another `BasePage` (or something like that) model which adds fields we think all of our subpage types are going to need, then inherit from `BasePage` instead of `Page`  in the actual project. Definitely doesn't need to be like that, fine to repeat the fields over 2 models, but if there you forsee all pages needing to share some fields that aren't in the default `Page` type then this could be a cool improvement!

If you do want to do that, I would suggest using an [Abstract Base Class ](https://docs.djangoproject.com/en/2.2/topics/db/models/#abstract-base-classes) as that will just add those fields to any model that inherits from `BasePage` in the table it creates in the DB. If you don't make it abstract then `BasePage` is stored separately as its own table.

**Comment by arh30** [comment]
Ah yes that sounds better @alistairclark ! Thanks for the helps and explanations, super helpful. Will have a go at this

File: {{cookiecutter.project_slug}}/apps/settings/models.py (hunk 2 above)
Comment (arh30) [reply {{cookiecutter.project_slug}}/apps/settings/models.py]: looked into this one, its too long to go on one

File: {{cookiecutter.project_slug}}/templates/includes/meta_wagtail.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  <meta name="twitter:description" content="{{ settings.settings.ShareMeta.default_share_description }}">
+  <meta itemprop="description" content="{{ settings.settings.ShareMeta.default_share_description }}">
+  <meta property="og:description" content="{{ settings.settings.ShareMeta.default_share_description }}" />
+{% endif %}
+
+{% if page.share_image %}
+  {% image page.share_image fill-1200x630 as share_image %}
+  <meta property="og:image" content="{{ share_image.url }}" />
+  <meta name="twitter:image" content="{{ request.build_absolute_uri }}{{ share_image.url }}">
+  <meta itemprop="image" content="{{ share_image.url }}">
```
[the remark this answers is not in the document]
Comment (arh30) [reply {{cookiecutter.project_slug}}/templates/includes/meta_wagtail.html]: hmm I thought the image tag was funny about using them outside of the if blocks, and usually needs to be used directly below... but i might be wrong

**Pull request merged**